pkg/validator/validator.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2024 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validator
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/okteto/okteto/pkg/env"
+)
+
+// ErrReservedVariableName is raised when a variable from cmd option has invalid name
+var ErrReservedVariableName = errors.New("Reserved variable name. See documentation for more info: https://www.okteto.com/docs/core/credentials/environment-variables/")
+
+// CheckReservedVariablesNameOption returns an error when any of the variable names from command flags are not allowed as input
+func CheckReservedVariablesNameOption(variables []string) error {
+	for _, v := range variables {
+		name, _, ok := strings.Cut(v, "=")
+		if ok && isReservedVariableName(name) {
+			return ErrReservedVariableName
```

Comment:
In different places of the CLI we always name the var causing the warning, for example when authenticating via env vars, we mention them. I imagine a complex command, where the user might not notice which var is reserved, it may be tricky or require further checks (ie. open the docs) to understand which one is reserved. Why not making it simpler by printing like:

```
 x  OKTETO_CONTEXT is a reserved variable name. See documentation for more info: https://www.okteto.com/docs/core/credentials/environment-variables/
 ```


```suggestion
			return fmt.Errorf("%s is a %w", name, ErrReservedVariableName)
```